packages/tempus/src/Tempus_StepperImplicit_impl.hpp

```diff
@@ -245,9 +218,19 @@ void StepperImplicit<Scalar>::setSolver(
        "Error - ModelEvaluator is unset!  Should call setModel() first.\n");
 
   solver_->setModel(wrapperModel_);
+  this->isInitialized_ = false;
+}
 
+
+template<class Scalar>
+void StepperImplicit<Scalar>::initialize()
+{
+  this->setSolver();
+  this->setObserver();
```

Comment:
I started out as you mentioned here, but then switched to adding these calls to handle the default situation where the app does not set them, but the initialize will at least set default values if they unset.  I do not have a problem going with what you suggest here.  I was just trying to handle situation of where the app wants to do the minimum amount of work and Tempus will fill it in with defaults.

Reply:
yeah - it's more a cosmetic thing that's caught my eye a few times in reviewing code.

Reply:
Fair.  I think we can have default observers set in the Stepper's constructors.  The issue was if the Stepper has a composite observer, you can not easily set a default observer and add observers to the composite without worrying about removing the default observer (another reason to go with a Stepper having an observer).  So if we go with a Stepper having an observer, we can set a default which the app can reset.  So the setObserver in the initialize could be removed.

The issue with setSolver is that there is a dependence on setModel.  I can put in some logic to call setSolver when setModel is called if the solver is set, and visa versa set the model on the solver if the model is set, and check everything in the initialize.  I will go this route, because like you said (and happens to me, indicating a point of confusion), "Every time I see this I have to stop and think about a set() call that doesn't take a argument."  I will likely go with setDefaultSolver() and setDefaultObserver().

Reply:
I also added changes that removed the confusing "setObserver()".  Now there is no default setting for setObserver().

I still need to look at doing the same for setSolver().

I did a force push on this branch after a "git rebase", so your branches probably need to be cleaned up.